### PR TITLE
WIP: Prefer forest labels over wilderness

### PR DESCRIPTION
**Pull request by zerebubuth**
**NOTE: This is only a partial PR at this point, and only starts to address parts of #1608.**

This was starting to take a while, so I wanted to check with you, @nvkelso, that it's heading in the right direction. And that the examples I'm using for the tests are appropriate. If it looks OK, then let me know and I'll continue porting across the rest of the rules (although perhaps trying to find a better way to say "name includes 'National Park'" that isn't so anglocentric).

Simplify most tier 1 & 2 POI min zooms to area-based thresholds, rather than formulae. This makes it easier for a human to predict what the output will be for a given input feature. Added new area thresholds for green areas which aren't national parks.

Connects to #1608.

### CODE REVIEW

**Comment by nvkelso** [comment]
> (although perhaps trying to find a better way to say "name includes 'National Park'" that isn't so anglocentric).

Looks like the example scene file YAML was before we started looking at `operator` and `protection_title` in the vector query. So either no change is needed, or you could look for more national park operators globally to add to the list (like we were doing for road shields)?

```
  - filter:
      boundary: national_park
      any:
        all:
          not: { operator: *us_forest_service }
          protect_class: ['2', '3', '5']
        operator: *us_parks_service
        operator:en: Parks Canada
        designation: national_park
        protection_title: National Park
    min_zoom: *tier1_min_zoom
    output:
      <<: *output_properties
      kind: national_park
      tier: 1
```

File: integration-test/1608-prefer-forest-labels-over-wilderness.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            }),
+        )
+
+        # TODO: should be national park, or nature_reserve? on the one hand,
+        # it is tagged as operated by NPS, but its name contains "recreation
+        # area", not "park"?
+        self.assert_has_feature(
+            z, x, y, 'pois', {
+                'id': 5908558,
+                'kind': 'national_park',
```
Comment (nvkelso) [inline integration-test/1608-prefer-forest-labels-over-wilderness.py]: I'd be fine with this being a `nature_reserve` instead. The US National Park Service operates many different units, but only a minority are truly national parks.

File: integration-test/473-landuse-tier.py [hunk 2]
```diff
@@ -4,30 +4,77 @@
...
+            z-3, x//8, y//8, 'landuse',
             {'kind': 'national_park', 'id': -1453306, 'tier': 1,
              'min_zoom': 3})
 
         self.assert_has_feature(
-            6, 12, 23, 'pois',
+            z, x, y, 'pois',
             {'kind': 'national_park', 'id': -1453306, 'tier': 1,
-             'min_zoom': 3.74})
+             'min_zoom': 6})
```
Comment (nvkelso) [inline integration-test/473-landuse-tier.py]: zoom **5** seems like better min_zoom than 6 for this crown park.

(zoom 3.74 was too early)

**Comment by nvkelso** [comment]
Yes, headed in good direction :)

**Comment by zerebubuth** [comment]
I think I've updated this PR to include all the stuff from the stylesheet, although perhaps spelled in a slightly different way. Please check through the examples and let me know if you think any have been misclassified or assigned the wrong zoom.

## Zoom 5

With the new min zoom functions, I'm getting the following 22 labels showing up at zoom 5:

 * [Northeast Greenland National Park](https://osm.org/relation/8515283)
 * [Wrangell–St. Elias National Park](https://osm.org/relation/6561033)
 * [Glacier Bay National Park](https://osm.org/relation/6077677)
 * [Katmai National Park](https://osm.org/relation/9090352)
 * [Jasper National Park](https://osm.org/relation/6395738)
 * [Parque Nacional Natural Sierra de Chiribiquete](https://osm.org/relation/6637001)
 * [Parque Nacional Alto Purús](https://osm.org/relation/3874631)
 * [Taman Nasional Gunung Leuser](https://osm.org/relation/4573902)
 * [Parque Nacional Kaa-Iya del Gran Chaco](https://osm.org/relation/7728779)
 * [Lemmenjoen kansallispuisto](https://osm.org/way/345599967)
 * [Death Valley National Park](https://osm.org/relation/174732)
 * [Banff National Park](https://osm.org/relation/6365995)
 * [Yellowstone National Park](https://osm.org/relation/1453306)
 * [Cairngorms National Park](https://osm.org/relation/1947603)
 * [Parque Nacional Cordillera Azul](https://osm.org/relation/548656)
 * [Hardangervidda nasjonalpark](https://osm.org/relation/6317425)
 * [Padjelanta](https://osm.org/relation/62326)
 * [Parc national d'Odzala-Kokoua](https://osm.org/relation/5885230)
 * [Sarek](https://osm.org/relation/62327)
 * [Nationalpark Schleswig-Holsteinisches Wattenmeer](https://osm.org/relation/1237758)
 * [Munga-Thirri National Park](https://osm.org/relation/6796363)
 * [Southwest National Park](https://osm.org/relation/6187193)

I've looked through each of these, and the only one that I think might be a little dodgy is [Nationalpark Schleswig-Holsteinisches Wattenmeer](https://osm.org/relation/1237758). However, all of its tags look okay and appropriate to a national park - it's just a marine national park. Short of intersecting all national parks with land, I'm not sure how to exclude this one. Perhaps it's okay that it shows up at 5?

## Zoom 6:

Here are the 112 which are at zoom 6 in (roughly) North America. There are a further 441 in the whole world, and I'll happily give you the list... it just seemed like too much to manually verify.

 * [Mojave National Preserve](https://osm.org/relation/175098)
 * [Glacier National Park](https://osm.org/relation/1242641)
 * [Mojave Trails National Monument](https://osm.org/relation/6150224)
 * [Olympic National Park](https://osm.org/relation/163769)
 * [Riding Mountain National Park](https://osm.org/way/185735773)
 * [Grand Canyon National Park](https://osm.org/relation/183377)
 * [Craters of the Moon National Monument](https://osm.org/relation/904337)
 * [Parque Nacional Ciénaga de Zapata](https://osm.org/way/361290872)
 * [Yosemite National Park](https://osm.org/relation/1643367)
 * [Joshua Tree National Park](https://osm.org/relation/6145614)
 * [Grand Canyon-Parashant National Monument](https://osm.org/relation/5765280)
 * [Basin and Range National Monument](https://osm.org/relation/5976391)
 * [Big Bend National Park](https://osm.org/relation/2625382)
 * [Upper Missouri River Breaks National Monument](https://osm.org/relation/6057961)
 * [Parque Nacional Laguna del Tigre](https://osm.org/relation/7330557)
 * [Parc national du Mont-Tremblant](https://osm.org/relation/8008857)
 * [Kings Canyon National Park](https://osm.org/relation/175061)
 * [Parc marin du Saguenay–Saint‑Laurent](https://osm.org/relation/8388138)
 * [North Cascades National Park](https://osm.org/relation/2421537)
 * [Sequoia National Park](https://osm.org/relation/175060)
 * [Parque Nacional Jardines de la Reina](https://osm.org/way/371542550)
 * [Craters of the Moon National Preserve](https://osm.org/relation/904336)
 * [Grand Teton National Park](https://osm.org/relation/1492555)
 * [Parque Nacional Cumbres de Monterrey](https://osm.org/relation/936111)
 * [Mount Rainier National Park](https://osm.org/relation/1399219)
 * [Sonoran Desert National Monument](https://osm.org/relation/5827166)
 * [North Cascades National Park](https://osm.org/relation/2421537)
 * [Rio Grande del Norte National Monument](https://osm.org/relation/5932647)
 * [Organ Pipe Cactus National Monument](https://osm.org/relation/5826991)
 * [Vermilion Cliffs National Monument](https://osm.org/relation/5752869)
 * [Rocky Mountain National Park](https://osm.org/relation/390960)
 * [Gold Butte National Monument](https://osm.org/relation/7577134)
 * [Big Bend Ranch State Park](https://osm.org/relation/4100800)
 * [Badlands National Park](https://osm.org/relation/5731065)
 * [Hanford Reach National Monument](https://osm.org/relation/6129791)
 * [Carrizo Plain National Monument](https://osm.org/relation/6181911)
 * [Dinosaur National Monument](https://osm.org/relation/5749153)
 * [Organ Mountains-Desert Peaks National Monument](https://osm.org/relation/1113594)
 * [Crater Lake National Park](https://osm.org/relation/1274747)
 * [Petrified Forest National Park](https://osm.org/relation/4640162)
 * [Organ Mountains-Desert Peaks National Monument](https://osm.org/relation/1113594)
 * [Parque Nacional Mirador-Río Azul](https://osm.org/relation/546716)
 * [Shenandoah National Park](https://osm.org/relation/5548542)
 * [Waterton Lakes National Park](https://osm.org/relation/6395476)
 * [Parc national de la Mauricie](https://osm.org/relation/8008852)
 * [Ironwood Forest National Monument](https://osm.org/relation/5827127)
 * [Berryessa Snow Mountain National Monument](https://osm.org/relation/6176314)
 * [Zion National Park](https://osm.org/relation/5758583)
 * [Canyonlands National Park - Island In The Sky District](https://osm.org/relation/5873932)
 * [White Sands National Monument](https://osm.org/relation/1504622)
 * [Canyonlands National Park - Needles District](https://osm.org/relation/5873934)
 * [Santa Rosa and San Jacinto Mountains National Monument](https://osm.org/relation/6140735)
 * [Biscayne National Park](https://osm.org/relation/9107419)
 * [Lassen Volcanic National Park](https://osm.org/relation/6207158)
 * [Great Sand Dunes National Park](https://osm.org/relation/5725308)
 * [El Malpais National Monument](https://osm.org/relation/5946885)
 * [Padre Island National Seashore](https://osm.org/way/614279786)
 * [Berryessa Snow Mountain National Monument](https://osm.org/relation/6176314)
 * [Parque Nacional Nevado de Toluca](https://osm.org/relation/936150)
 * [Craters of the Moon National Preserve](https://osm.org/relation/904336)
 * [Canyon de Chelly National Monument](https://osm.org/relation/5813412)
 * [Valles Caldera National Preserve](https://osm.org/relation/171095)
 * [Channel Islands National Park](https://osm.org/relation/3636334)
 * [Parque Nacional Guanahacabibes](https://osm.org/relation/6488943)
 * [Great Basin National Park](https://osm.org/relation/5974781)
 * [Redwood National Park](https://osm.org/relation/215231)
 * [Arches National Park](https://osm.org/relation/5868384)
 * [Guadalupe Mountains National Park](https://osm.org/relation/5953502)
 * [Canyonlands National Park - Maze District](https://osm.org/relation/5873933)
 * [Point Reyes National Seashore](https://osm.org/relation/1250137)
 * [New River Gorge National River](https://osm.org/relation/3178915)
 * [Cascade-Siskiyou National Monument](https://osm.org/relation/6094810)
 * [Agua Fria National Monument](https://osm.org/relation/5812470)
 * [Organ Mountains-Desert Peaks National Monument](https://osm.org/relation/1113594)
 * [Theodore Roosevelt National Park South Unit](https://osm.org/way/60510385)
 * [Saguaro National Park (East) - Rincon Mountain District](https://osm.org/relation/5821126)
 * [Parque Nacional Desembarco del Granma](https://osm.org/way/361219689)
 * [Channel Islands National Park](https://osm.org/relation/3636334)
 * [Parc national des Monts-Valin](https://osm.org/relation/8388316)
 * [Reserva de la biosfera La Michilía](https://osm.org/way/553655462)
 * [Mesa Verde National Park](https://osm.org/relation/5749837)
 * [Berryessa Snow Mountain National Monument](https://osm.org/relation/6176314)
 * [Lava Beds National Monument](https://osm.org/relation/6208245)
 * [Mammoth Cave National Park](https://osm.org/way/220416402)
 * [Parc national du Fjord-du-Saguenay](https://osm.org/relation/8388140)
 * [Great Sand Dunes National Preserve](https://osm.org/relation/5725309)
 * [Parque Nacional El Tepozteco](https://osm.org/relation/930900)
 * [Parque Nacional Pico La Bayamesa](https://osm.org/way/371542552)
 * [Carlsbad Caverns National Park](https://osm.org/relation/5953459)
 * [Parque Nacional Turquino](https://osm.org/way/361219690)
 * [Parc national de Frontenac](https://osm.org/relation/6175298)
 * [Olympic National Park](https://osm.org/relation/163769)
 * [Parc national du Fjord-du-Saguenay](https://osm.org/relation/8388140)
 * [Bryce Canyon National Park](https://osm.org/relation/4716641)
 * [Mojave National Preserve](https://osm.org/relation/175098)
 * [Wind Cave National Park](https://osm.org/relation/2998186)
 * [Wupatki National Monument](https://osm.org/relation/5813556)
 * [Theodore Roosevelt National Park North Unit](https://osm.org/way/60510306)
 * [Parque Nacional Pico Cristal](https://osm.org/way/361220610)
 * [Parc national de la Pointe-Taillon](https://osm.org/relation/7295219)
 * [Black Canyon of the Gunnison National Park](https://osm.org/relation/395552)
 * [Bandelier National Monument](https://osm.org/relation/2923958)
 * [Chaco Culture National Historical Park](https://osm.org/relation/1463209)
 * [Badlands National Park](https://osm.org/relation/5731065)
 * [Cuyahoga Valley National Park](https://osm.org/relation/1286297)
 * [John D. Rockefeller, Jr. Memorial Parkway](https://osm.org/relation/1453305)
 * [Channel Islands National Park](https://osm.org/relation/3636334)
 * [Pinnacles National Park](https://osm.org/relation/6183160)
 * [Acadia National Park](https://osm.org/relation/2176999)
 * [Sand to Snow National Monument](https://osm.org/relation/6138177)
 * [Cumberland Gap National Historical Park](https://osm.org/relation/7192241)
 * [Mojave Trails National Monument](https://osm.org/relation/6150224)

## Zoom distribution

Table of `min_zoom` by kind.

kind | z5 | z6 | z7 | z8 | z9 | z10 | z11 | z12 | z13
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
national_park | 22 | 553 | 70 | 562 | 198 | 581 | 250 | 313 | 243
protected_area |   |   | 652 | 592 | 373 | 1134 | 628 | 675 | 919
forest |   |   |   | 1291 | 558 | 19031 | 7249 | 8282 | 10121
park |   |   |   | 1011 | 1449 | 1712 | 2351 | 3058 | 34365
nature_reserve |   |   |   |   | 7001 | 5174 | 5321 | 6740 | 28248

Cumulative features at each zoom:

kind | z5 | z6 | z7 | z8 | z9 | z10 | z11 | z12 | z13
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
national_park | 22 | 575 | 645 | 1207 | 1405 | 1986 | 2236 | 2549 | 2792
protected_area | | | 652 | 1244 | 1617 | 2751 | 3379 | 4054 | 4973
forest | | | | 1291 | 1849 | 20880 | 28129 | 36411 | 46532
park | | | | 1011 | 2460 | 4172 | 6523 | 9581 | 43946
nature_reserve | | | | | 7001 | 12175 | 17496 | 24236 | 52484

**Pull request merged**